File: CarConfigurator.BL/Models/CarModel.cs
namespace CarConfigurator.BL.Models
{
    public class CarModel : IProduct
    {
        public string Name { get; }
        public string Description { get; }
        public decimal BasePrice { get; }
        public string EAN { get; set; }

        public CarModel(string ean, string name, string description, decimal basePrice)
        {
            EAN = ean;
            Name = name;
            Description = description;
            BasePrice = basePrice;
        }
    }
}

File: CarConfigurator/ClientApp/src/app/models/CarModel.ts
export class CarModel {
  private _name: string;
  private _description: string;

  public get name(): string {
    return this._name;
  }
  public get description(): string {
    return this._description;
  }

  constructor(name: string, description: string) {
    this._name = name;
    this._description = description;
  }
}
